<launch>

    <arg name="world" default="maze"/>
    <arg name="robot_count" default="2"/>
    <arg name="run" default="0"/>
    <arg name="method" default="gvg"/>
    <arg name="bs_pose" default="0,0"/>
    <arg name="environment"  default=''/>

    <param name="use_sim_time" value="true" />

    <node name="Stage" pkg="stage_ros" type="stageros" args="$(find gvgexploration)/world/$(arg world).world">
        <param name="base_watchdog_timeout" value="0" />
    </node>

    <param name="robot_count" value="$(arg robot_count)" />

    <group ns="robot_0">
        <arg name="robot_id" value="1"/>

        <include file="$(find gvgexploration)/launch/robot_sim.launch">
            <arg name="robot_id" value="$(arg robot_id)"/>
        </include>

        <include file="$(find gvgexploration)/launch/robot.launch">
            <arg name="robot_name" value="0" />
            <arg name="robot_type" value="2" />
            <arg name="base_stations" value="1" />

            <arg name="method" value="$(arg method)"/>
            <arg name="robot_count" value="$(arg robot_count)"/>
            <arg name="run" value="$(arg run)"/>
            <arg name="bs_pose" value="$(arg bs_pose)"/>
            <arg name="environment" value="$(arg environment)"/>
        </include>

    </group>

    <group ns="robot_1">
        <arg name="robot_id" value="2"/>
        <include file="$(find gvgexploration)/launch/robot_sim.launch">
            <arg name="robot_id" value="$(arg robot_id)"/>
        </include>

        <include file="$(find gvgexploration)/launch/robot.launch">
            <arg name="robot_name" value="1" />
            <arg name="robot_type" value="2" />
            <arg name="base_stations" value="0" />

            <arg name="method" value="$(arg method)"/>
            <arg name="robot_count" value="$(arg robot_count)"/>
            <arg name="run" value="$(arg run)"/>
            <arg name="bs_pose" value="$(arg bs_pose)"/>
            <arg name="environment" value="$(arg environment)"/>

        </include>
    </group>

    <node name="roscbt" pkg="gvgexploration" type="roscbt.py" respawn="true">
        <rosparam file="$(find gvgexploration)/param/roscbt_config2.yaml"/>
        <rosparam file="$(find gvgexploration)/param/gvg_config.yaml"/>
        <param name="method" value="$(arg method)"/>
        <param name="robot_count" value="$(arg robot_count)"/>
        <param name="run" value="$(arg run)"/>
        <param name="bs_pose" value="$(arg bs_pose)"/>
        <param name="environment" value="$(arg environment)"/>
    </node>

    <!-- RVIZ to view the visualization -->
    <node name="RVIZ" pkg="rviz" type="rviz" args=" -d $(find gvgexploration)/param/maze.rviz" output="screen" />

</launch>

